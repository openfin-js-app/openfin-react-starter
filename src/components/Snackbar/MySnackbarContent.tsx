import * as React from 'react';
import { MouseEventHandler } from "react";
import cx from 'classnames';

import SnackbarContent from '@material-ui/core/SnackbarContent';
import {SnackbarContentProps} from "@material-ui/core/SnackbarContent/SnackbarContent";
import IconButton from '@material-ui/core/IconButton';

import CheckCircleIcon from '@material-ui/icons/CheckCircle';
import DesktopMacIcon from '@material-ui/icons/DesktopMac';
import ErrorIcon from '@material-ui/icons/Error';
import InfoIcon from '@material-ui/icons/Info';
import CloseIcon from '@material-ui/icons/Close';
import WarningIcon from '@material-ui/icons/Warning';
import FaceIcon from '@material-ui/icons/Face'

import { WithStyles, withStyles} from '@material-ui/core/styles';
import { snackbarContentCompStyle as style } from '../../assets/jss/openfin-starter';

const variantIcon ={
    primary: DesktopMacIcon,
    success: CheckCircleIcon,
    warning: WarningIcon,
    error: ErrorIcon,
    info: InfoIcon,
    rose: FaceIcon,
};

interface IMySnackbarContentProps{
    classes?:any,
    className?:string,
    message:string,
    onClose:MouseEventHandler<any>,
    variant:string,
}

type Props = IMySnackbarContentProps & WithStyles<typeof style> & SnackbarContentProps

class MySnackbarContentComp extends React.Component<Props,{}>{
    render(){

        const {
            classes, className, message, onClose, variant, ...other
        } = this.props;

        const Icon = variantIcon[variant];

        return(
            <SnackbarContent
                className={cx( classes.outmostContent, classes[variant],className)}
                aria-describedby={"client-snackbar"}
                message={
                    <span id={"client-snackbar"} className={classes.message}>
                        <Icon className={cx( classes.icon, classes.iconVariant )} />
                        {message}
                    </span>
                }
                action={
                    <IconButton
                        key={"close"}
                        aria-label={"Close"}
                        color={"inherit"}
                        className={classes.close}
                        onClick={onClose}
                    >
                        <CloseIcon className={classes.icon}/>
                    </IconButton>
                }
                {...other}
            />
        );
    }
}

export default withStyles(style)(MySnackbarContentComp);