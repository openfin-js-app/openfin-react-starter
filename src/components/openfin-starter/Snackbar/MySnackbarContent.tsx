import * as React from 'react';
import { MouseEventHandler } from "react";
import cx from 'classnames';

import SnackbarContent from '@material-ui/core/SnackbarContent';
import {SnackbarContentProps} from "@material-ui/core/SnackbarContent/SnackbarContent";
import IconButton from '@material-ui/core/IconButton';

import CheckCircleIcon from '@material-ui/icons/CheckCircle';
import DesktopMacIcon from '@material-ui/icons/DesktopMac';
import ErrorIcon from '@material-ui/icons/Error';
import InfoIcon from '@material-ui/icons/Info';
import CloseIcon from '@material-ui/icons/Close';
import WarningIcon from '@material-ui/icons/Warning';
import FaceIcon from '@material-ui/icons/Face'

import { makeStyles } from '@material-ui/styles';
import { snackbarContentCompStyle as style } from '../../../assets/jss/openfin-starter';

const variantIcon ={
    primary: DesktopMacIcon,
    success: CheckCircleIcon,
    warning: WarningIcon,
    error: ErrorIcon,
    info: InfoIcon,
    rose: FaceIcon,
};

interface IMySnackbarContentProps{
    classes?:any,
    className?:string,
    message:string,
    onClose:MouseEventHandler<any>,
    variant:string,
}

type Props = IMySnackbarContentProps & SnackbarContentProps

const useStyles = makeStyles(style);

const MySnackbarContentComp:React.FunctionComponent<Props> = (
    {
        className, message, onClose, variant, ...other
    }
)=>{

    const classes = useStyles();

    const Icon = variantIcon[variant];

    return(
        <SnackbarContent
            className={cx( classes.outmostContent, classes[variant],className)}
            aria-describedby={"client-snackbar"}
            message={
                <span id={"client-snackbar"} className={classes.message}>
                        <Icon className={cx( classes.icon, classes.iconVariant )} />
                    {message}
                    </span>
            }
            action={
                <IconButton
                    key={"close"}
                    aria-label={"Close"}
                    color={"inherit"}
                    onClick={onClose}
                >
                    <CloseIcon className={classes.icon}/>
                </IconButton>
            }
            {...other}
        />
    );

}

export default MySnackbarContentComp;