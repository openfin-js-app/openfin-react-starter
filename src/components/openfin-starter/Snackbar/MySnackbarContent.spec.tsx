import * as React from 'react';
import toJson from 'enzyme-to-json';
import { createShallow, createMount } from '@material-ui/core/test-utils';

import MySnackbarContent from './MySnackbarContent';

describe('MySnackbarContent comp',()=>{

    let shallow;
    let mount;

    beforeEach(() => {
        shallow = createShallow();
        mount = createMount();
    });

    afterEach(()=>{
        mount.cleanUp();
    })

    it('renders correctly by default',()=>{

        const onClose = jest.fn()

        const wrapper = shallow(<MySnackbarContent
            message={'message'}
            onClose={onClose}
            variant={'primary'}
        />);

        const component = wrapper.dive();
        expect(toJson(component)).toMatchSnapshot();

    })

});