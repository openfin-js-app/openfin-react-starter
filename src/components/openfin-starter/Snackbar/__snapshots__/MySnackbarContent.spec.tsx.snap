// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MySnackbarContent comp renders correctly by default 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
